### PR TITLE
Bugfix. Top-level constants should not be considered magic number

### LINKED ISSUE #841: Top-level constants should not be considered magic number
## Describe the bug
```kotlin
package com.example

private const val FOO = 42

class Example
```

## Environment information
* diktat version: 0.5.0

### PR DESCRIPTION
**Pull request by aktsay6**
### What's done:
  * Fixed bug

This pull request closes #841

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt [hunk 1]
```diff
@@ -37,11 +38,19 @@ class MagicNumberRule(configRules: List<RulesConfig>) : DiktatRule(
         )
     }
     override fun logic(node: ASTNode) {
-        if (node.elementType == INTEGER_CONSTANT || node.elementType == FLOAT_CONSTANT) {
+        if ((node.elementType == INTEGER_CONSTANT || node.elementType == FLOAT_CONSTANT)
+                && !isTopLevelDeclarationConstant(node)) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt:42]: I believe this should be checked along with `isConstant` parameter; you mix the internal logic of the rule when extracting some of the checks to a higher level

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt [hunk 2]
```diff
@@ -47,7 +48,7 @@ class MagicNumberRule(configRules: List<RulesConfig>) : DiktatRule(
         val nodeText = node.treePrev?.let { if (it.elementType == OPERATION_REFERENCE && it.hasChildOfType(MINUS)) "-${node.text}" else node.text } ?: node.text
         val isIgnoreNumber = configuration.ignoreNumbers.contains(nodeText)
         val isHashFunction = node.parent({ it.elementType == FUN && it.isHashFun() }) != null
-        val isPropertyDeclaration = node.parent({ it.elementType == PROPERTY && !it.isNodeFromCompanionObject() }) != null
+        val isPropertyDeclaration = node.parent({ it.elementType == PROPERTY && !it.isNodeFromCompanionObject() && !isTopLevelDeclarationConstant(it) }) != null
         val isLocalVariable = node.parent({ it.isVarProperty() && (it.psi as KtProperty).isLocal }) != null
         val isConstant = node.parent({ it.elementType == PROPERTY && it.isConstant() }) != null
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt:53]: Why are top-level constants not handled correctly with this setting?

Comment (aktsay6) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt:53]: Reworked logic of excluding const properties.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt [hunk 3]
```diff
@@ -47,7 +48,7 @@ class MagicNumberRule(configRules: List<RulesConfig>) : DiktatRule(
         val nodeText = node.treePrev?.let { if (it.elementType == OPERATION_REFERENCE && it.hasChildOfType(MINUS)) "-${node.text}" else node.text } ?: node.text
         val isIgnoreNumber = configuration.ignoreNumbers.contains(nodeText)
         val isHashFunction = node.parent({ it.elementType == FUN && it.isHashFun() }) != null
-        val isPropertyDeclaration = node.parent({ it.elementType == PROPERTY && !it.isNodeFromCompanionObject() }) != null
+        var isPropertyDeclaration = node.parent({ it.elementType == PROPERTY && !it.isNodeFromCompanionObject() }) != null
         val isLocalVariable = node.parent({ it.isVarProperty() && (it.psi as KtProperty).isLocal }) != null
         val isConstant = node.parent({ it.elementType == PROPERTY && it.isConstant() }) != null
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt:53]:
```suggestion
        val isConstant = node.parent({ it.elementType == PROPERTY && it.isConstant() }) != null
        val isPropertyDeclaration = !isConstant && node.parent({ it.elementType == PROPERTY && !it.isNodeFromCompanionObject() }) != null
        val isLocalVariable = node.parent({ it.isVarProperty() && (it.psi as KtProperty).isLocal }) != null
```
Why not like this, keeping both variables imutable?

**Review by petertrr — APPROVED** [review]

**Pull request merged**